### PR TITLE
Remove remaining uses of `childToSlot`

### PR DESCRIPTION
**Pull request by tvolkert**
## Description

The remaining uses of this pattern were all due to wanting to have
the child's slot when `Element.forgetChild()` was called. However,
when that method is called, the child's `slot` value is still valid
in the context of the parent, so the uses can just use `child.slot`.

## Related Issues

This is the final round of cleanup from the fallout of #63269

## Tests

Test exemption: this PR just removes code (and renames a few variables) - the existing tests passing validates the change.

## Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*

### CODE REVIEW

**Comment by tvolkert** [comment]
Adding @goderbauer for context of the change (since he reviewed the earlier changes)

Adding @HansMuller and @justinmc for the specific Material changes.

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/cupertino/text_selection.dart [hunk 1]
```diff
@@ -747,8 +747,7 @@ class _CupertinoTextSelectionToolbarItemsElement extends RenderObjectElement {
   ) : super(widget);
 
   List<Element> _children;
-  final Map<_CupertinoTextSelectionToolbarItemsSlot, Element> slotToChild = <_CupertinoTextSelectionToolbarItemsSlot, Element>{};
-  final Map<Element, _CupertinoTextSelectionToolbarItemsSlot> childToSlot = <Element, _CupertinoTextSelectionToolbarItemsSlot>{};
+  final Map<_CupertinoTextSelectionToolbarItemsSlot, Element> slottedChildren = <_CupertinoTextSelectionToolbarItemsSlot, Element>{};
```
Comment (HansMuller) [inline packages/flutter/lib/src/cupertino/text_selection.dart:750]: This name sounds like a list rather than a map (slotToChild sounds like a map)

**Review by HansMuller — APPROVED** [review]
LGTM modulo (my humble opinion) renaming the slotToChild properties.

File: packages/flutter/lib/src/material/chip.dart [hunk 2]
```diff
@@ -2066,8 +2066,7 @@ enum _ChipSlot {
 class _RenderChipElement extends RenderObjectElement {
   _RenderChipElement(_ChipRenderWidget chip) : super(chip);
 
-  final Map<_ChipSlot, Element> slotToChild = <_ChipSlot, Element>{};
-  final Map<Element, _ChipSlot> childToSlot = <Element, _ChipSlot>{};
+  final Map<_ChipSlot, Element> children = <_ChipSlot, Element>{};
```
Comment (justinmc) [inline packages/flutter/lib/src/material/chip.dart:2069]: Why is this `children` here but `slottedChildren` in the cupertino file above?

File: packages/flutter/lib/src/material/chip.dart [hunk 3]
```diff
@@ -2077,29 +2076,26 @@ class _RenderChipElement extends RenderObjectElement {
...
   }
 
   @override
   void forgetChild(Element child) {
-    assert(slotToChild.values.contains(child));
-    assert(childToSlot.keys.contains(child));
-    final _ChipSlot slot = childToSlot[child];
-    childToSlot.remove(child);
-    slotToChild.remove(slot);
+    assert(children.values.contains(child));
```
Comment (justinmc) [inline packages/flutter/lib/src/material/chip.dart:2084]: Nit: I know you're just changing the variable name, but couldn't this be:

```suggestion
    assert(children.containsValue(child));
```

Here and in a couple other places below.

**Review by justinmc — APPROVED** [review]
LGTM, just a naming question 👍

**Pull request merged**